src/components/services-view/services-view.controller.ts

```diff
@@ -1,156 +1,173 @@
+    public openOrderingPanel() {
+        this.ctrl.orderingPanelvisible = true;
+    };
+
+    public closeOrderingPanel = () => {
+        this.ctrl.orderingPanelvisible = false;
+    };
+
+    private updateAll() {
+        this.updateCatalogServices();
+        this.updateOpenshiftImages();
+    }
+
+    private updateState() {
+        this.ctrl.loading = _.isEmpty(this.ctrl.catalogServices) || _.isEmpty(this.ctrl.openshiftImageStreams);
+
+        if (!this.ctrl.loading) {
+            this.ctrl.filteredItems = this.ctrl.allItems;
+        }
     }
-  }
-
-  public getSubCategories(category: string) {
-    let subCats = [{id: 'all', label:  'All'}];
-    this.ctrl.categories.map(categoryObj => {
-      if (category === 'all' || category === categoryObj.id) {
-        subCats = subCats.concat(categoryObj.subCategories);
-      }
-    });
-
-    subCats = this.makeRows(subCats);
-
-    return subCats;
-  };
-
-  public handleClick(item: any, e: any) {
-    item.ctrl.serviceToOrder = item;
-    item.ctrl.openOrderingPanel();
-  };
-
-  public openOrderingPanel() {
-    this.ctrl.orderingPanelvisible = true;
-  };
-
-  public closeOrderingPanel = () => {
-    this.ctrl.orderingPanelvisible = false;
-  };
-
-  private updateAll() {
-    this.updateCategories();
-    this.updateServices();
-  }
-
-  private updateServices() {
-    this.ctrl.origServices = this.normalizeServices();
-    this.ctrl.filteredServices = this.ctrl.origServices;
-  }
-
-  private updateCategories() {
-    this.ctrl.categories = this.ctrl.categories;
-    this.ctrl.subCategories = this.getSubCategories('all');
-  }
-
-  private makeRows(subCats: any) {
-    let subCatsRows: any = [];
-    for (let i = 0, len = subCats.length; i < len; i += this.subCatMaxRowLength) {
-      subCatsRows.push(subCats.slice(i, i + this.subCatMaxRowLength));
+
+    private updateCatalogServices() {
+        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('service', this.ctrl.catalogServices));
+        this.updateState();
+    }
+
+    private updateOpenshiftImages() {
+        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('image', this.ctrl.openshiftImageStreams));
+        this.updateState();
     }
-    return subCatsRows;
-  }
-
-  private getRowOfSubCategory(subCategory: string) {
-    for (let row = 0; row < this.ctrl.subCategories.length; row += 1) {
-      for (let card = 0; card < this.ctrl.subCategories[row].length; card += 1) {
-        if (this.ctrl.subCategories[row][card].id === subCategory) {
-          return row;
+
+    private makeRows(subCats: any) {
+        let subCatsRows: any = [];
+        for (let i = 0, len = subCats.length; i < len; i += this.subCatMaxRowLength) {
+            subCatsRows.push(subCats.slice(i, i + this.subCatMaxRowLength));
         }
-      }
+        return subCatsRows;
+    }
+
+    private getRowOfSubCategory(subCategory: string) {
+        for (let row = 0; row < this.ctrl.subCategories.length; row += 1) {
```

Comment:
i know you said this code is changing, but a general comment is that we use _.each instead of for loops, the code is cleaner to read without all the array indexing